File: examples/allow-and-deny/allow-and-deny.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Statement } from '../../lib';

function getStatements() {
  function wrap() {
    // doc-start
    const s1 = new Statement.Ec2() //
      .allow()
      .toStartInstances();

    const s2 = new Statement.Ec2() //
      .deny()
      .toStopInstances();
    // doc-end
    return [s1, s2];
  }
  return wrap();
}
const s = getStatements();
out(s);
deploy(s);

File: examples/allow-and-deny/allow-and-deny.py
import json
import iam_floyd as statement


def statements():
    # doc-start
    s1 = statement.Ec2() \
                .allow() \
                .to_start_instances()

    s2 = statement.Ec2() \
                .deny() \
                .to_stop_instances()
    # doc-end
    return [s1, s2]


for s in statements():
    pretty = json.dumps(s.to_json(), indent=4, sort_keys=True)
    print(pretty)
